lib/ansible/module_utils/network/eos/config/interfaces/interfaces.py

```diff
@@ -0,0 +1,250 @@
+        result['commands'] = commands
+
+        changed_interfaces_facts = self.get_interfaces_facts()
+
+        result['before'] = existing_interfaces_facts
+        if result['changed']:
+            result['after'] = changed_interfaces_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_interfaces_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_interfaces_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = _param_list_to_dict(want)
+        have = _param_list_to_dict(have)
+        state = self._module.params['state']
+        if state == 'overridden':
+            commands = self._state_overridden(want, have)
+        elif state == 'deleted':
+            commands = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands = self._state_merged(want, have)
+        elif state == 'replaced':
+            commands = self._state_replaced(want, have)
+        return commands
+
+    @staticmethod
+    def _state_replaced(want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = _compute_commands(want, have, replace=True, remove=True)
+
+        replace = commands['replace']
+        remove = commands['remove']
+
+        commands_by_interface = replace
+        for interface, commands in remove.items():
+            commands_by_interface[interface] = replace.get(interface, []) + commands
+
+        return _flatten_commands(commands_by_interface)
+
+    @staticmethod
+    def _state_overridden(want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        # Add empty desired state for unspecified interfaces
+        for key in have:
+            if key not in want:
+                want[key] = {}
+
+        # Otherwise it's the same as replaced
+        return Interfaces._state_replaced(want, have)
```

Comment:
Can _state_* be changed to the class methods as it invokes methods from within the class? I don't see a reason to have these methods as` staticmethod`

Reply:
+1 on ganesh's comment here